Libraries/LibC/string.cpp

```diff
@@ -457,9 +457,50 @@ char* strpbrk(const char* s, const char* accept)
 
 char* strtok(char* str, const char* delim)
 {
-    (void)str;
-    (void)delim;
-    ASSERT_NOT_REACHED();
+    static char* saved_str;
+    
+    if (!str) {
+        if (!saved_str)
+            return nullptr;
+        str = saved_str;
+    }
+    
+    size_t token_start = 0;
+    size_t token_end = 0;
+    
+    for (size_t i = 0; i < strlen(str); ++i) {
```

Comment:
Please don't use `strlen()` in loop conditions, as they will be re-evaluated on each iteration.
Instead, cache the result of `strlen()` outside of the loop, and compare against the cached result.

Reply:
Shouldn't the compiler hoist those expressions outside of the loops in one of the optimization passes?

Reply:
It might. But we should not write O(n^2) code and hope for the compiler to save us when it's trivial to avoid :)

Reply:
Fair point.